backend/coins/eth/rpcclient/wrapper/wrapper.go

```diff
@@ -0,0 +1,146 @@
+package wrapper
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/BitBoxSwiss/bitbox-wallet-app/backend/accounts"
+	"github.com/BitBoxSwiss/bitbox-wallet-app/backend/coins/eth"
+	"github.com/BitBoxSwiss/bitbox-wallet-app/backend/coins/eth/rpcclient"
+
+	"github.com/BitBoxSwiss/bitbox-wallet-app/backend/coins/eth/erc20"
+	ethtypes "github.com/BitBoxSwiss/bitbox-wallet-app/backend/coins/eth/types"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type rpcClientWithTransactionsSource interface {
+	eth.TransactionsSource
+	rpcclient.Interface
+}
+
+type client struct {
+	defaultRpcClient  rpcClientWithTransactionsSource
+	fallbackRpcClient rpcClientWithTransactionsSource
+
+	// TODO do we maybe want to add some fields to keep track
+	// of which one we are using? if we switch to the fallback one
+	// it might make sense to use that for a while instead of wasting
+	// time each time trying the default first and falling back to the
+	// other one later.
+}
+
+// NewClient creates a new client that uses the default RPC client unless it fails,
+// in which case it falls back to the fallback RPC client.
+func NewClient(defaultRPCClient, fallbackRPCClient rpcClientWithTransactionsSource) *client {
```

Comment:
It's unusual to see an exported method return an instance an unexported struct. I'm somewhat surprised it even compiles :D